telemetry_core/src/main/java/com/newrelic/telemetry/logs/Log.java

```diff
@@ -0,0 +1,194 @@
+  private final String logType;
+  private final String level;
+
+  private Log(LogBuilder builder) {
+    this.timestamp = builder.timestamp;
+    this.message = builder.message;
+    this.attributes = builder.attributes;
+    this.serviceName = builder.serviceName;
+    this.logType = builder.logType;
+    this.level = builder.level;
+  }
+
+  public long getTimestamp() {
+    return timestamp;
+  }
+
+  public String getMessage() {
+    return message;
+  }
+
+  public Attributes getAttributes() {
+    return attributes;
+  }
+
+  public String getServiceName() {
+    return serviceName;
+  }
+
+  public String getLogType() {
+    return logType;
+  }
+
+  public String getLevel() {
+    return level;
+  }
+
+  public static LogBuilder builder() {
+    return new LogBuilder();
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof Log)) {
+      return false;
+    }
+
+    Log log = (Log) o;
+
+    if (timestamp != log.timestamp) {
+      return false;
+    }
+    if (message != null ? !message.equals(log.message) : log.message != null) {
+      return false;
+    }
+    if (attributes != null ? !attributes.equals(log.attributes) : log.attributes != null) {
+      return false;
+    }
+    if (serviceName != null ? !serviceName.equals(log.serviceName) : log.serviceName != null) {
+      return false;
+    }
+    if (logType != null ? !logType.equals(log.logType) : log.logType != null) {
+      return false;
+    }
+    return level != null ? level.equals(log.level) : log.level == null;
+  }
+
+  @Override
+  public int hashCode() {
+    int result = (int) (timestamp ^ (timestamp >>> 32));
+    result = 31 * result + (message != null ? message.hashCode() : 0);
+    result = 31 * result + (attributes != null ? attributes.hashCode() : 0);
+    result = 31 * result + (serviceName != null ? serviceName.hashCode() : 0);
+    result = 31 * result + (logType != null ? logType.hashCode() : 0);
+    result = 31 * result + (level != null ? level.hashCode() : 0);
+    return result;
+  }
+
+  @Override
+  public String toString() {
+    return "Log{"
+        + "timestamp="
+        + timestamp
+        + ", message='"
+        + message
+        + '\''
+        + ", attributes="
+        + attributes
+        + ", serviceName='"
+        + serviceName
+        + '\''
+        + ", logType='"
+        + logType
+        + '\''
+        + ", level='"
+        + level
+        + '\''
+        + '}';
+  }
+
+  /**
+   * A class for holding the variables associated with a Span object and creating a new Span object
+   * with those variables
+   */
+  public static class LogBuilder {
+    private long timestamp = System.currentTimeMillis();
+    private String message;
+    private Attributes attributes = new Attributes();
+    private String serviceName; // service.name <- goes in attributes
+    private String logType; // logtype <- goes in attributes
```

Comment:
I would strike this; I don't think it adds value being a separate attribute.

Reply:
I believe it does add value. It makes it very easy for people to add the logtype attribute, which is described in the logging API docs. Making the semantic conventions easy to use it a definitely value-add.

Reply:
this has been removed, because there are no formal semantic conventions defined for it.